ng2-material/components/menu/menu.ts

```diff
@@ -0,0 +1,125 @@
+import {Component, ViewEncapsulation, ElementRef, Renderer} from "angular2/core";
+import {DOM} from "angular2/src/platform/dom/dom_adapter";
+
+
+@Component({
+  selector: 'md-menu',
+  template: `<ng-content></ng-content>`,  
+  encapsulation: ViewEncapsulation.None
+})
+export class MdMenu {
+  private _menuElement: HTMLElement;
+  private _triggerElement: Element;
+  private _menuContent: Element;
+  private _menuContainers: any;
+  private _isOpen: boolean = false;
+
+  constructor(private _element: ElementRef, private _renderer: Renderer) {
+    this._menuElement = this._element.nativeElement;
+  }
+
+  ngAfterContentInit() {
+    if (this._menuElement.children.length != 2) {
+      throw Error('Expected two children elements.');
+    }
+
+    this.initMenu();
+  }
+
+  initMenu() {
+    var body = DOM.query('body');
+    
+    this._renderer.setElementClass(this._menuElement, 'md-menu', true);
+    this._renderer.setElementClass(this._menuElement, '_md', true);
+
+    this._triggerElement = this._menuElement.firstElementChild;
+
+    if (this._triggerElement && (
+      this._triggerElement.nodeName == 'MD-BUTTON' ||
+      this._triggerElement.nodeName == 'BUTTON'
+    ) && !this._triggerElement.hasAttribute('type')) {
+      this._triggerElement.setAttribute('type', 'button');
+    }
+
+    this._triggerElement && this._triggerElement.setAttribute('aria-haspopup', 'true');
+
+    this._menuContent = this._menuElement.querySelector('md-menu-content');
+
+    for (var i = 0; i < (<HTMLElement>this._menuContent).children.length; i++) {
+      let menuItem = (<HTMLElement>this._menuContent).children[i];
+      if (menuItem.nodeName != 'MD-MENU-ITEM') continue;
+      if ((<HTMLElement>menuItem).children.length != 1) continue;
+      let button = (<HTMLElement>menuItem).children[0];
+      this._renderer.setElementStyle(button, 'text-align', 'left');
+      
+      this._renderer.listen(button, 'click', (event) => {
+        this.closeMenu();
+      });
+
+    }
+
+
+    this._menuContainers = this._renderer.createElement(body,'div');
```

Comment:
May be I'm wrong but I think you should avoir DOM direct manipulation. There's  a lot of stuffs in Angular that allow you to manipulate it via Angular features.